go/proxyd/latestblock.go

```diff
@@ -0,0 +1,61 @@
+package proxyd
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+type LatestBlockHead struct {
+	url  string
+	quit chan struct{}
+
+	mutex sync.Mutex
+	head  *types.Header
+}
+
+func newLatestBlockHead(url string) *LatestBlockHead {
+	return &LatestBlockHead{
+		url:  url,
+		quit: make(chan struct{}),
+	}
+}
+
+func (h *LatestBlockHead) Start() error {
+	client, err := ethclient.DialContext(context.Background(), h.url)
```

Comment:
Once we switch to a polling loop, let's definitely make sure that we have some kind of exponential backoff in case the block number backend goes down.

Reply:
The `ethclient` module handles a couple details with connection management which simplifies the code. But I get your general point here for polling. Will make the change.